### PR TITLE
Scrutinizer score

### LINKED ISSUE #675: Improve Scrutinizer Score
They provide a lot of great code improvement fixes that we should be completing ongoing as we can on Scrutinizer: https://scrutinizer-ci.com/g/WordImpress/Give/code-structure/master/hot-spots

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

This pr will fix #675 

## Types of changes

Refactor code.

#### Refactor
1. `give_process_paypal_payment` function refactored into:
     1. `give_validate_nonce`
     2. `_give_build_paypal_url` further refoctored into:
        1. _`give_get_paypal_button_type`

#### New 
1. `give_check_variable` https://github.com/WordImpress/Give/commit/c491622e1c1cdfa3b55fb1293c3d7205dc0227c4
     

## Checklist:
- [ ] My code is tested
        (PayPal sandbox is not loading properly).
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by ravinderk** [comment]
@DevinWalker we have an older version on Travis:

screen shot 2016-12-08 at 11 01 30 pm

Current version is `5.7`

File: tests/unit-tests/tests-formatting.php [hunk 1]
```diff
@@ -210,4 +210,21 @@ public function give_let_to_num_provider() {
...
+	 *
+	 * @cover  give_validate_nonce
+	 */
+	function test_give_validate_nonce() {
+		$input_nonce = wp_create_nonce( 'give_gateway' );
+
+		/*
+		 * If nonce does not validate successfully then WPDieException throw.
+		 */
+		$this->expectException( 'WPDieException' );
```
Comment (DevinWalker) [inline tests/unit-tests/tests-formatting.php]: @ravinderk I'm seeing some failing unit tests for this, can you check into: https://travis-ci.org/WordImpress/Give/builds/182321052

File: includes/gateways/functions.php [hunk 2]
```diff
@@ -327,4 +329,173 @@ function give_get_ordered_payment_gateways( $gateways ) {
...
+
+/**
+ * Get paypal button type.
+ *
+ * Note: only for internal use
+ *
+ * @since 1.8
+ * @return string
+ */
+function _give_get_paypal_button_type() {
```
Comment (DevinWalker) [inline includes/gateways/functions.php]: Why are you prefixing with underscores?

File: includes/formatting.php [hunk 3]
```diff
@@ -473,3 +473,68 @@ function give_let_to_num( $size ) {
...
+ *
+ * @since 1.8
+ *
+ * @param                   $variable
+ * @param string (optional) $conditional , default value: isset
+ * @param bool (optional)   $default     , default value: false
+ *
+ * @return mixed
+ */
+function give_check_variable( $variable, $conditional = '', $default = false ) {
```
Comment (DevinWalker) [inline includes/formatting.php]: Are you using this anywhere?

File: includes/gateways/functions.php (hunk 2 above)
Comment (ravinderk) [inline includes/gateways/functions.php]: @DevinWalker  I thought this function should be only used for internal purpose.

File: includes/formatting.php (hunk 3 above)
Comment (ravinderk) [inline includes/formatting.php]: @DevinWalker I forgot to use that but here is example https://github.com/WordImpress/Give/pull/1325/commits/79fbaa5cbadca086d22816ceaa918f7ade2210f0

File: includes/gateways/functions.php [hunk 4]
```diff
@@ -327,4 +329,173 @@ function give_get_ordered_payment_gateways( $gateways ) {
...
+ * Build paypal url
+ *
+ * Note: For internal use only.
+ *
+ * @param int   $payment_id   Payment ID
+ * @param array $payment_data Array of payment data.
+ *
+ * @return mixed|string
+ */
+function _give_build_paypal_url( $payment_id, $payment_data ) {
```
Comment (DevinWalker) [inline includes/gateways/functions.php]: Can we more this into the `gateways/paypal-standard.php` file? Also, the underscore seems not very useful unless we implement it across the project. I'd say lost the leading `_` of the function.

File: includes/gateways/functions.php (hunk 2 above)
Comment (DevinWalker) [inline includes/gateways/functions.php]: Same as above. Please move into gateway file and remove underscore.

**Comment by ravinderk** [comment]
@DevinWalker pr updated.

**Pull request merged**